### PR TITLE
Google Vertex API

### PR DESCRIPTION
**Pull request by NeonBohdan**
# Description
Uses Google API to generate responces

# Issues

# Other Notes

### CODE REVIEW

File: neon_llm_palm2/palm2.py [hunk 1]
```diff
@@ -150,7 +164,8 @@ def _embeddings(self, question: str, answers: List[str], persona: dict) -> (List
         """
         response = self.ask(question, [], persona=persona)
         texts = [response] + answers
-        embeddings = get_embeddings(texts, engine="text-embedding-ada-002")
+        embeddings_obj = self._embedding.get_embeddings(texts)
```
Comment (NeonDaniel) [inline neon_llm_palm2/palm2.py]: Should this reference `self.embedding` to ensure `self._embedding` is defined?

File: requirements/requirements.txt [hunk 2]
```diff
@@ -1,4 +1,5 @@
 # model
+google-cloud-aiplatform
 openai[embeddings]~=0.27
```
Comment (NeonDaniel) [inline requirements/requirements.txt:3]: Is this dependency still used, or just leftover from ChatGPT? I don't see any remaining references

Comment (NeonBohdan) [reply requirements/requirements.txt:3]: Its used for embeddings comparison 
If needed can use other lib with less dependencies

File: neon_llm_palm2/palm2.py (hunk 1 above)
Comment (NeonBohdan) [reply neon_llm_palm2/palm2.py]: Done with https://github.com/NeonGeckoCom/neon-llm-palm2/pull/1/commits/d3e3e06aca7a2af53e77ef697a3fa8e03cbf3a3a

**Review by NeonDaniel — APPROVED** [review]

**Pull request merged**